### PR TITLE
Fixed deprecation warning in sprockets 3.7.0. Fixes #415

### LINKED ISSUE #415: Deprecation warning in sprockets 3.7.0
Using i18n-js 3.0.0.rc13, sprockets 3.7.0 and Rails 4.2.7:

```bash
DEPRECATION WARNING: You are using the a deprecated processor interface #<Proc:0x007fe3aac61228@/Users/jsantos/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/i18n-js-3.0.0.rc13/lib/i18n/js/engine.rb:30>.
Please update your processor interface:
https://github.com/rails/sprockets/blob/master/guides/extending_sprockets.md#supporting-all-versions-of-sprockets-in-processors
 (called from block (2 levels) in <class:Engine> at /Users/jsantos/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/i18n-js-3.0.0.rc13/lib/i18n/js/engine.rb:30)
```

This has been recently fixed in other gems, such as [sass-rails](https://github.com/rails/sass-rails/issues/381#issuecomment-234733694).

### PR DESCRIPTION
**Pull request by mamantoha**

### CODE REVIEW

**Comment by PikachuEXE** [comment]
This PR only remove the warning but does not make the gem support Sprockets 4
I have seen PR for supporting sprockets 4 in https://github.com/railsware/js-routes/pull/202
Would you like to study it and change this PR to do similar thing?

If not I can try to do that when free. But I got limited time.

**Comment by mamantoha** [comment]
@PikachuEXE thanks. I will take a look.

**Comment by mamantoha** [comment]
I've tested it with Sprockets `3.7` and `4.0.0.beta2`. It's working.

**Comment by PikachuEXE** [comment]
Thanks for changing the code in a short time.
Let me try it (I am using sprockets 2) and I will merge it once tested.

**Comment by PikachuEXE** [comment]
The cache expiration works fine in development
sprockets (2.12.4)
sprockets-rails (2.3.3)
rails (4.1.16)

**Comment by fwininger** [comment]
Hi guys !
Thanks @mamantoha and @PikachuEXE to tackle this issue.

I have try the branch with Ruby 2.3.1, Rails 5.0 and sprockets 3.7.0, but I have always the same warning...

I have have in my Gemfile : 

```
gem 'i18n-js', '>= 3.0.0.rc13', git: 'https://github.com/mamantoha/i18n-js.git'
```

So in my Gemfile.lock : 

```
+  GIT
+  remote: https://github.com/mamantoha/i18n-js.git
+  revision: 53cc1b49923e3e22e3ceefe5bc774d2460abe30d
+  specs:
+    i18n-js (3.0.0.rc13)
+      i18n (~> 0.6, >= 0.6.6)
```

After with rails c : 

```
DEPRECATION WARNING: You are using the a deprecated processor interface #<Proc:0x000000021d44d8@/home/user/.rvm/gems/ruby-2.3.1/bundler/gems/i18n-js-53cc1b49923e/lib/i18n/js/engine.rb:75>.
Please update your processor interface:
https://github.com/rails/sprockets/blob/master/guides/extending_sprockets.md#supporting-all-versions-of-sprockets-in-processors
 (called from block (2 levels) in <class:Engine> at /home/user/.rvm/gems/ruby-2.3.1/bundler/gems/i18n-js-53cc1b49923e/lib/i18n/js/engine.rb:75)
```

Thanks

File: lib/i18n/js/engine.rb [hunk 1]
```diff
@@ -1,8 +1,51 @@
...
+      v3 = Gem::Dependency.new('', ' ~> 3')
+      v4 = Gem::Dependency.new('', ' >= 4')
+
+      sprockets_version = Gem::Version.new(Sprockets::VERSION).release
+
+      initializer_args  = case sprockets_version
+                          when -> (v) { v2.match?('', v) }
+                            { after: "sprockets.environment" }
+                          when -> (v) { v3.match?('', v) || v4.match?('', v) }
+                            { after: :engines_blank_point, before: :finisher_hook, silence_deprecation: true }
```
Comment (PikachuEXE) [inline lib/i18n/js/engine.rb]: No `else`?

**Comment by PikachuEXE** [comment]
@fwininger 
This PR just follow what https://github.com/railsware/js-routes/pull/202 does
Maybe you try that gem and see if you got same deprecation warning?

File: lib/i18n/js/engine.rb (hunk 1 above)
Comment (mamantoha) [inline lib/i18n/js/engine.rb]: I forgot to add `this`

**Comment by mamantoha** [comment]
@fwininger this warning should not appear with Sprockets 3.7.0. Anyway, I will take a look at this later.

**Comment by PikachuEXE** [comment]
By reading https://github.com/rails/sprockets/blob/master/guides/extending_sprockets.md#supporting-all-versions-of-sprockets-in-processors

I think we should use the new `I18nSprocketsExtension` for all 3 sprockets versions
@mamantoha 
Can you change the PR again? Thanks :)

**Comment by mamantoha** [comment]
I've changed the PR again. I'm not sure that it works with Sprockets 2. Please check. Works fine with 2 and 3. I think need more testing from other users.

**Comment by fwininger** [comment]
Hi !

Sorry for the delay, I retried this week-end but I don't have the Deprecation Warning anymore.
It's a lite bite strange because I don't have the warning with this PR and also we the current version.
I can't reproduce the state with the deprecation warning...

Anyway the current version and this PR works for me with Ruby 2.3.1 and Rails 5.0.

**Comment by PikachuEXE** [comment]
@fwininger 
The deprecation warning only shows for sprockets version latest 3.x I think (read that from their change log)

I will try once with my sprockets 2.x tomorrow

**Comment by mamantoha** [comment]
The deprecation warning only shows for Sprockets version latest 3.7

**Comment by fwininger** [comment]
I use Sprockets version 3.7

```
    rails (5.0.0.1)
      actioncable (= 5.0.0.1)
      actionmailer (= 5.0.0.1)
      actionpack (= 5.0.0.1)
      actionview (= 5.0.0.1)
      activejob (= 5.0.0.1)
      activemodel (= 5.0.0.1)
      activerecord (= 5.0.0.1)
      activesupport (= 5.0.0.1)
      bundler (>= 1.3.0, < 2.0)
      railties (= 5.0.0.1)
      sprockets-rails (>= 2.0.0)
    sprockets (3.7.0)
      concurrent-ruby (~> 1.0)
      rack (> 1, < 3)
    sprockets-rails (3.1.1)
      actionpack (>= 4.0)
      activesupport (>= 4.0)
      sprockets (>= 3.0.0)
    i18n (0.7.0)
    i18n-js (3.0.0.rc13)
      i18n (~> 0.6, >= 0.6.6)
    sass-rails (5.0.6)
      railties (>= 4.0.0, < 6)
      sass (~> 3.1)
      sprockets (>= 2.8, < 4.0)
      sprockets-rails (>= 2.0, < 4.0)
      tilt (>= 1.1, < 3)
```

**Comment by PikachuEXE** [comment]
After the last change 
The cache does not expire when translation file content changed
Tested locally
Using 
- sprockets (2.12.4)

**Comment by joker-777** [comment]
Hi, we use sprockets 3.7.0 and this branch works like expected.

**Comment by PikachuEXE** [comment]
@mamantoha 
Made some changes to make it work on sprockets 2.x
And opened #418

@joker-777 @fwininger 
Please test again with that PR, thx

**Comment by PikachuEXE** [comment]
I think the code that makes this PR bugged in sprockets 2.x is `after: "sprockets.environment"`
Which was fixed but reverted in here incorrectly

**Comment by mamantoha** [comment]
@PikachuEXE great 👍 .

confirming that #418 works on version 3.7.
I close this PR.

**Comment by PikachuEXE** [comment]
@mamantoha 
You are so fast o_o

**Comment by mamantoha** [comment]
@PikachuEXE I have a small app for test purposes :) I just switched to that PR and tested whether it works for me

**Pull request closed without merging**